### PR TITLE
Update for k8s secrets

### PR DESCRIPTION
**Pull request by barleyj-opengov**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: incubator/rundeck/values.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ replicaCount: 1
 
 image:
   repository: rundeck/rundeck
-  tag: 3.0.16
+  tag: 3.0.23
```
Comment (Chili-Man) [inline incubator/rundeck/values.yaml]: why not 3.0.26 or 3.1.0

File: incubator/rundeck/templates/rundeck-environment-configmap.yaml [hunk 2]
```diff
@@ -2,6 +2,5 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: {{ .Release.Name }}-environment-configmap
-type: Opaque
 data:
-{{ toYaml .Values.rundeck.env | indent 4}}
\ No newline at end of file
+{{ toYaml .Values.rundeck.env | indent 4}}
```
Comment (Chili-Man) [inline incubator/rundeck/templates/rundeck-environment-configmap.yaml]: why did you remove the `type: Opaque`

File: incubator/rundeck/values.yaml (hunk 1 above)
Comment (barleyj-opengov) [reply incubator/rundeck/values.yaml]: 3.0.23 was what we tested with. The others may be newer, but we haven't tried them yet.

File: incubator/rundeck/templates/rundeck-environment-configmap.yaml (hunk 2 above)
Comment (barleyj-opengov) [reply incubator/rundeck/templates/rundeck-environment-configmap.yaml]: It was causing a problem when trying to deploy it initially. Let me try adding it back in.
